poll_manager.cpp

```diff
@@ -163,16 +163,9 @@ void PollManager::apsdeDataConfirm(const deCONZ::ApsDataConfirm &conf)
         return;
     }
 
-    if (dstAddr.hasExt() && conf.dstAddress().hasExt()
-        && dstAddr.ext() != conf.dstAddress().ext())
+    if (!isSameAddress(dstAddr, conf.dstAddress()))
     {
-
-    }
-
-    else if (dstAddr.hasNwk() && conf.dstAddress().hasNwk()
-        && dstAddr.nwk() != conf.dstAddress().nwk())
-    {
-
+        return;
```

Comment:
This is an interesting spot, I think the reason why I didn't add a `return;` back in the day was that the `confirm.id()` already uniquely matches the confirm to the request. In theory we can ditch the address check here altogether, but it doesn't hurt and in worst case would run in a small timeout and polling keeps going.